src/app/Home.tsx

```diff
@@ -29,11 +29,12 @@ export default function Home({ session }: HomeProps) {
   const dropQuery = useQuery(GetDrop);
   const collection = dropQuery.data?.drop.collection;
   const metadataJson = collection?.metadataJson;
+  const walletAddresses = me?.wallets?.map((wallet) => wallet?.address);
   const holder = useMemo(() => {
-    return collection?.holders?.find(
-      (holder: Holder) => holder.address === me?.wallet?.address
+    return collection?.holders?.find((holder: Holder) =>
+      walletAddresses?.includes(holder.address)
     );
-  }, [collection?.holders, me?.wallet]);
+  }, [collection?.holders, walletAddresses]);
```

Comment:
We need to refactor this to work off of the purchases instead of holders since if the user transfers asset they are no longer a holder but shouldn't let them mint again since they already did so. The purchase history is better because it has who minted it originally.